woocommerce/woocommerce-paypal-payments
# Google Pay: Fix the incorrect popup triggering (2645)

### Description

This PR fixes the incorrect Google Pay behavior upon user button click. Our async transaction info fetching was being interpreted by browsers as 'non-user initiated' resulting in a "`To make this purchase using Google Pay, open this website in a browser app, such as Chrome or Safari`" error.

This PR adds transaction info preloading, resolving the issue.


### Steps to Test

1. Activate Google Pay.
2. Add a product to the cart.
3. Try to pay with Google Pay, make sure you get the correct screen, and confirm that you can complete the order.

### Screenshots

|Before|After|
|-|-|
|![Google Pay 2](https://github.com/user-attachments/assets/b577e9eb-f28e-4fe7-a4b0-e229974e80ce)|![Google_Pay](https://github.com/user-attachments/assets/72ff7c3c-205a-4fd3-8a34-dfa93941c277)|
